node/hybrid-node-tests/commonjs-test/commonjs-test.cjs

```diff
@@ -0,0 +1,53 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const { AsyncClient } = require("glide-rs");
+const RedisServer = require("redis-server");
+const FreePort = require("find-free-port");
+const PORT_NUMBER = 4001;
+
+let server;
+let port;
+
+const { exec } = require("child_process");
+
+function flushallOnPort(port) {
+    return new Promise((resolve, reject) => {
+        exec(`redis-cli -p ${port} FLUSHALL`, (error, _, stderr) => {
+            if (error) {
+                console.error(stderr);
+                reject(error);
+            } else {
+                resolve();
+            }
+        });
+    });
+}
+
+FreePort(PORT_NUMBER)
+    .then(([free_port]) => {
+        port = free_port;
+        server = new RedisServer(port);
+        server.open(async (err) => {
+            if (err) {
+                console.error("Error opening server:", err);
+                throw err;
+            }
+
+            const client = AsyncClient.CreateConnection(
+                `redis://localhost:${port}`,
+            );
+            await client.set("test", "test");
+            let result = await client.get("test");
+
+            if (result !== "test") {
+                throw new Error("Common Test failed");
+            } else {
+                console.log("Common Test passed");
+            }
+
+            await flushallOnPort(port);
+            await server.close();
```

Comment:
node has its weirdness about top-level async functions. did you run this with console logs, and verify that all of this happened before the script completes?